lib/runner/extensions/request.command.js

```diff
@@ -1,52 +1,78 @@
 var _ = require('lodash'),
     async = require('async'),
+    uuid = require('uuid'),
     sdk = require('postman-collection'),
 
-    // we need to access tough-cookie if no external cookie jar is provided. however, the `request` module does not work
-    // with vanilla tough-cookie and needs a flavour of tough-cookie returned by itself. Hence we depend on
-    // `postman-request` here.
-    postmanRequest = require('postman-request'),
+    // These are functions which a request passes through _before_ being sent. They take care of stuff such as
+    // variable resolution, loading of files, etc.
+    prehelpers = require('../request-helpers-presend'),
 
-    helpers = require('../request-helpers'),
+    // Similarly, these run after the request, and have the power to dictate whether a request should be re-queued
+    posthelpers = require('../request-helpers-postsend'),
 
-    Requester = require('../../requester').Requester;
+    Authorizer = require('../../authorizer').Authorizer,
+    Requester = require('../../requester').Requester,
 
-module.exports = {
-    init: function (done) {
-        var timeout = _.min([
-                _.get(this.options, 'timeout.request'),
-                _.get(this.options, 'timeout.global')
-            ]),
-            cookieJar;
-
-        // @todo - remove this when requester creation is offloaded to runner
-        if (this.options.requester && this.options.requester.external === true) {
-            this.requester = this.options.requester.instance;
-            return done();
-        }
+    CONTEXT_PROPERTIES = [
+        'data',
+        'environment',
+        'globals',
+        'coords'
```

Comment:
yep, that way, let's say if you prompt the user for basic auth password, it's re-used.